flink-core/src/main/java/org/apache/flink/configuration/WebOptions.java

```diff
@@ -156,6 +156,48 @@
                     .withDeprecatedKeys("jobmanager.web.backpressure.delay-between-samples")
                     .withDescription("This config option is no longer used");
 
+    /** Time, in milliseconds, after which cached stats are cleaned up if not accessed. */
```

Comment:
I understand the intent, but I think it might add some potential for confusion with respect to how the sampling works. Mainly because window semantics in Flink is based on time, whereas here it is about gathering a certain number of data points. Not sure it is fits.

Reply:
You also mentioned offline that this is similar to how backpressure sampling works. So if a user already knows that, it's better to keep it in this way.

Reply:
Yes, this approach is largely inspired by the previously existing backpressure monitor which was also based on stack traces sampling. A couple of pointers:
https://github.com/apache/flink/blob/c685251ce3b39362e45b07a12aa0cc6e7f665021/flin[…]re/src/main/java/org/apache/flink/configuration/WebOptions.java
https://github.com/apache/flink/blob/7e4694b8198664300b13b5304c62271b192d4512/flin[…]pache/flink/runtime/webmonitor/StackTraceSampleCoordinator.java
https://github.com/apache/flink/blob/7e4694b8198664300b13b5304c62271b192d4512/flin[…]g/apache/flink/runtime/webmonitor/BackPressureStatsTracker.java